aten/src/ATen/FunctionalTensorWrapper.cpp

```diff
@@ -42,6 +42,10 @@ void FunctionalTensorWrapper::set_constructor_metadata() {
   // TODO: metadata copying may not actually be necessary then
   set_custom_sizes_strides(SizesStridesPolicy::CustomSizes);
   set_custom_device(true);
+  // E.g. when running torch.compile under inference mode, we need to make sure that
+  // for any inputs that were created outside of inference mode (so they are not inference tensors),
+  // then the functional wrappers that we wrap them with should also not be inference tensors.
+  version_counter_ = value_.unsafeGetTensorImpl()->version_counter();
```

Comment:
Wouldn't this access to the version_counter raise an error on inference Tensors?

Reply:
Talked offline - we're not accessing the version counter, just straight up copying the struct onto the wrapper.

Also - we copy the dispatch keyset from the inner tensor onto the wrapper, so if the inner tensor has the Autograd dispatch key (because it was created outside of inference mode), then the wrapper will as well (even though it was created in inference mode).